components/camel-splunk-hec/src/main/java/org/apache/camel/component/splunkhec/SplunkHECConfiguration.java

```diff
@@ -99,6 +101,20 @@ public void setHost(String host) {
         this.host = host;
     }
 
+    /**
+     * Splunk endpoint
+     *   Defaults to /services/collector/event
+     *   To write RAW data like JSON use /services/collector/raw
+     *   To extract timestamps in Splunk>8.0 /services/collector/event?auto_extract_timestamp=true
+     */
+    public void setEndpoint(String endpoint) {
```

Comment:
Are there other endpoints? And endpoint can maybe confuse as Camel also has a concept of endpoint. So maybe we can come up with a different name?

For the last with the query parameter then that will not work as Camel think its another parameter.

So maybe come up

1) a new name, with 3 enum values: default, raw, timestamp
2) And then you can set the url based on the endpoint inside the camel component

Reply:
Hi Claus, thank you very much for your quick reply and check of the PR!

- Thank you very mich for the hint with the name, okay to set it to "SplunkRESTEndpoint"? Which would be a better naming. 

- If the url includes a question mark the user can utilize RAW{} https://camel.apache.org/manual/faq/how-do-i-configure-endpoints.html#HowdoIconfigureendpoints-Configuringparametervaluesusingrawvalues so it will work. Modules like azure-eventhub do make excessive use of it for connection strings, passwords, ... for example:

`.from(azure-eventhubs:?connectionString=RAW({{camel.component.azure-eventhubs.connection-string}}) ...`

- For the reason of extendability, enum does not sound like a good fit to me, since it requires changes to the software and different versions in place on every new Splunk version/endpoint. The ENUM may also suggest RESTEndpoints that are not available in older Splunk versions. So I suggest to stick with a strong default that does not introduce a breaking change, but full flexibility to adpot to new Splunk versions without changing the camel module. See: https://docs.splunk.com/Documentation/SplunkCloud/8.2.2203/Data/HECRESTendpoints

- Naming in the module can be improved a lot, since hostAndIp are written aus SplunkURL where users normally expect an URL compliant to RFC1738 (https://datatracker.ietf.org/doc/html/rfc1738) including the REST endpoint including http or https (including RAW() for handling slashes). One possible way to improve and keep backwards compability would be introduction of a configuration parameter "connectionString" that has first priority if set and includes all information. But I guess that is better adressed in a second pull request.

Reply:
Thanks for the detailed feedback. Yeah components can surely be improved, you are welcome to send more PRs.